eslint-plugin-expensify/tests/prefer-at.test.js

```diff
@@ -0,0 +1,110 @@
+const RuleTester = require('@typescript-eslint/rule-tester').RuleTester;
+const rule = require('../prefer-at');
+const message = require('../CONST').MESSAGE.PREFER_AT;
+
+const ruleTester = new RuleTester({
+    parser: '@typescript-eslint/parser',
+    parserOptions: {
+        project: './tsconfig.json',
+        tsconfigRootDir: __dirname,
+        sourceType: 'module',
+        ecmaVersion: 2020,
+    },
+});
+
+ruleTester.run('prefer-at', rule, {
+    valid: [
+        {
+            code: 'const example = [1, 2, 3, 4]; example.at(0)',
+        },
+        {
+            code: 'const test = [1, 2, 3, 4]; test.at(1)',
+        },
+        {
+            code: 'const sample = [1, 2, 3, 4]; sample.at(-1)',
+        },
+        {
+            code: 'const testing = [1, 2, 3, 4]; testing.at(-2)',
+        },
+        {
+            code: 'const testing = [[1, 2], [3, 4]]; testing.at(0).at(-1)',
+        },
+        {
+            code: 'const test = [1, 2, 3, 4]; test.at((test.length - 1) / 2)',
+        },
+        {
+            code: `
+                const object = {0: 'v0', 1: 'v1', 2: 'v2'};
+                object[0]
+                `,
+        },
```

Comment:
```suggestion
        },
        {
            code: `
                const index = 1;
                const object = {0: 'v0', 1: 'v1', 2: 'v2'};
                object[index]
                `,
        },
```

Could we add this test case too?